helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/TopStateMaxCapacityUsageInstanceConstraint.java

```diff
@@ -39,6 +41,14 @@ protected double getAssignmentScore(AssignableNode node, AssignableReplica repli
       // So return zero on any assignable node candidate.
       return 0;
     }
+
+    // No node or replica capacity defined, we default to just looking at # top state
+    if (node.getMaxCapacity().isEmpty() && replica.getCapacity().isEmpty()) {
+      int curTopPartitionCountForResource = node.getAssignedTopStatePartitionsCount();
+      int estimatedMaxTopStateCount = clusterContext.getEstimatedMaxTopStateCount();
+      return computeUtilizationScore(estimatedMaxTopStateCount, curTopPartitionCountForResource);
+    }
```

Comment:
Will this be the right behavior? If Capacity is not defined, we should not process and let the computation failed because this is an error.

Adding this code is more like a hiding the problem.

Expected behavior should be : 
1. Evenness of top state should be handled in different constraint. 
2. If it miss the required field, we should fail the computation and let user handle it.

Reply:
Currently users can onboard to WAGED without setting instance and partition capacities. WAGED validation fails if only 1 has been defined or if there are missing keys, but will successfully run if neither are defined. In this case, WAGED doesn't ensure any top state evenness as the assignment scores will always be 0 and tiebreak will be determined by the node's logicalID.

Should the expectation that users have to define at least a default instance and partition capacity for the cluster? Like so:
```
  "mapFields": {
    "DEFAULT_INSTANCE_CAPACITY_MAP": {
      "PARTCOUNT": "2000",
    },
   "DEFAULT_PARTITION_WEIGHT_MAP": {
      "PARTCOUNT": "386"
    }

...

  "listFields": {
    "INSTANCE_CAPACITY_KEYS": [
      "PARTCOUNT"
    ]
```

Reply:
Yeah. I think my second point get the answer. But still we should not make top state evenness into capacity constraint. Logically it is not clear. 

Would be good to have a separate constraints to hold it.